lean4-infoview/src/infoview/messages.tsx

```diff
@@ -188,13 +188,14 @@ export function useMessagesForFile(uri: DocumentUri, line?: number): Interactive
                 if (err?.code === -32801) {
                     // Document has been changed since we made the request.
                     // This can happen while typing quickly, so server will catch up on next edit.
+                    console.log('getInteractiveDiagnostics: document has been changed since we made the request ')
                 } else {
                     console.log('getInteractiveDiagnostics error ', err)
                 }
             }
         }
     }
-    React.useEffect(() => void updateDiags(), [uri, line, lspDiags.get(uri)])
+    React.useEffect(() => void updateDiags(), [uri, line, rs, lspDiags.get(uri)])
```

Comment:
Thanks for catching this! I think we should use `rs.sessionIdAt(uri)` here instead, it was made specifically to serve as a dep but I forgot to actually use it :| It might be missing in other places.

Reply:
Thanks, I switched to this, it seems to work.  I also checked AllMessages and it already had the "rs" dependency which is probably why AllMessages was fine.